src/plugins/intel_cpu/src/nodes/grid_sample.cpp

```diff
@@ -0,0 +1,277 @@
+            p.srcWidthMul2F  = std::vector<float>(jitKernel->getDataElPerVec(), p.srcWidthF  * 2.f);
+            if (alignCorners) {
+                p.srcHeightMul2Sub1F = std::vector<float>(jitKernel->getDataElPerVec(), p.srcHeightSub1F[0] * 2.f);
+                p.srcWidthMul2Sub1F  = std::vector<float>(jitKernel->getDataElPerVec(), p.srcWidthSub1F[0]  * 2.f);
+                p.wDenormCoefF = std::vector<float>(jitKernel->getDataElPerVec(), (p.srcWidthF  - 1.f) / 2.f);
+                p.hDenormCoefF = std::vector<float>(jitKernel->getDataElPerVec(), (p.srcHeightF - 1.f) / 2.f);
+            } else {
+                p.srcHeightMul2Sub1F = std::vector<float>(jitKernel->getDataElPerVec(), p.srcHeightMul2F[0] - 1.f);
+                p.srcWidthMul2Sub1F  = std::vector<float>(jitKernel->getDataElPerVec(), p.srcWidthMul2F[0]  - 1.f);
+            }
+        }
+        p.srcChannelStepB = srcDataShape[2] * srcDataShape[3] * dataTypeSize;
+        p.dstChannelStepB = dstShape[2] * dstShape[3] * dataTypeSize;
+    });
+}
+
+void GridSample::execute(dnnl::stream strm) {
+    const void* srcData = getParentEdgeAt(IN_DATA)->getMemoryPtr()->GetPtr();
+    const uint8_t* gridData = reinterpret_cast<uint8_t*>(getParentEdgeAt(IN_GRID)->getMemoryPtr()->GetPtr());
+    uint8_t* dstData = reinterpret_cast<uint8_t*>(getChildEdgeAt(0)->getMemoryPtr()->GetPtr());
+
+    auto threadBody = [&](const int ithr, const int nthr) {
+        const auto& p = execParamsPerThread[ithr];
+        auto arg = jGridSamplesExecArgs();
+
+        arg.src                = srcData;
+        arg.grid               = gridData + p.gridStartB;
+        arg.dst                = dstData  + p.dstStartB;
+        arg.batchNum           = p.batchNum;
+        arg.channelsNum        = p.channelsNum;
+        arg.srcHeightF         = &p.srcHeightF;
+        arg.srcWidthF          = &p.srcWidthF;
+        arg.srcWidthB          = &p.srcWidthB;
+        arg.srcChannelStepB    = p.srcChannelStepB;
+        arg.dstChannelStepB    = p.dstChannelStepB;
+        arg.srcBatchStepB      = p.srcBatchStepB;
+        arg.gridBatchStepB     = p.gridBatchStepB;
+        arg.dstBatchStepB      = p.dstBatchStepB;
+        arg.srcHeightSub1F     = p.srcHeightSub1F.data();
+        arg.srcWidthSub1F      = p.srcWidthSub1F.data();
+        arg.srcWidthMul2F      = p.srcWidthMul2F.data();
+        arg.srcHeightMul2F     = p.srcHeightMul2F.data();
+        arg.srcHeightMul2Sub1F = p.srcHeightMul2Sub1F.data();
+        arg.srcWidthMul2Sub1F  = p.srcWidthMul2Sub1F.data();
+        arg.wDenormCoefF       = p.wDenormCoefF.data();
+        arg.hDenormCoefF       = p.hDenormCoefF.data();
+        arg.workAmount         = p.workAmount;
+
+        (*jitKernel)(&arg);
+    };
+
+    parallel_nt(0, threadBody);
+}
+
+void GridSample::executeDynamicImpl(dnnl::stream strm) {
+    execute(strm);
+}
+
+std::vector<VectorDims> GridSample::shapeInfer() const {
+    return Node::shapeInferGeneric(PortMask(1, 2));
```

Comment:
If I missed nothing, `2` is extra because we have only 2 inputs: `0` port - Data and `1` port - Grid. Could you please verify?